packages/backend/server/src/__tests__/utils/testing-app.ts

```diff
@@ -1,37 +1,38 @@
-import {
-  ConsoleLogger,
-  INestApplication,
-  ModuleMetadata,
-} from '@nestjs/common';
+import { INestApplication, ModuleMetadata } from '@nestjs/common';
+import type { NestExpressApplication } from '@nestjs/platform-express';
 import { TestingModuleBuilder } from '@nestjs/testing';
 import { User } from '@prisma/client';
 import cookieParser from 'cookie-parser';
 import graphqlUploadExpress from 'graphql-upload/graphqlUploadExpress.mjs';
 import supertest from 'supertest';
 
-import { ApplyType, GlobalExceptionFilter } from '../../base';
+import { AFFiNELogger, ApplyType, GlobalExceptionFilter } from '../../base';
 import { AuthService } from '../../core/auth';
 import { UserModel } from '../../models';
 import { createTestingModule } from './testing-module';
 import { initTestingDB, TEST_LOG_LEVEL } from './utils';
-interface TestingAppMeatdata extends ModuleMetadata {
+
+interface TestingAppMetadata extends ModuleMetadata {
   tapModule?(m: TestingModuleBuilder): void;
   tapApp?(app: INestApplication): void;
 }
 
 export type TestUser = Omit<User, 'password'> & { password: string };
 
 export async function createTestingApp(
-  moduleDef: TestingAppMeatdata = {}
+  moduleDef: TestingAppMetadata = {}
 ): Promise<TestingApp> {
   const module = await createTestingModule(moduleDef, false);
 
-  const app = module.createNestApplication({
+  const app = module.createNestApplication<NestExpressApplication>({
     cors: true,
     bodyParser: true,
     rawBody: true,
   });
-  const logger = new ConsoleLogger();
+  if (AFFiNE.flavor.doc) {
+    app.useBodyParser('raw');
```

Comment:
what's the difference with `{ bodyParser: true }` in creatingApp

Reply:
This is 2 separate configurations. `rawBody` requires `bodyParser` to be enabled to have this capability.
Setting `raw` here is to enable `bodyParser` to handle request `application/octet-stream` data type, which is disabled by default.
We need to enable it on doc service.

https://github.com/expressjs/body-parser/blob/master/lib/types/raw.js#L40